chromium_src/chrome/browser/ui/webui/searchbox/realbox_handler.cc

```diff
@@ -15,7 +17,32 @@
   autocomplete_input.set_prefer_keyword(true)
 #define set_allow_exact_keyword_match(allow) set_allow_exact_keyword_match(true)
 
+// Unfortunately, plumbing through the source doesn't seem trivial - it looks
+// like it should be possible with the {source} part but it seems like it only
+// works with Google Search. Additionally, the {source} param treats the NTP
+// Realbox and the Omnibox the same, which isn't ideal, so we'd have to patch
+// the behavior there too. This seems like the patch of least changes:
+// 1. If this is a keyword search with Brave Search
+// 2. Then replace &source=desktop with &source=new_tab
+#define OpenURL(params, callback)                                             \
+  GetURL();                                                                   \
+  GURL url = destination_url;                                                 \
+  auto* turl =                                                                \
+      GetTemplateURLService()->GetTemplateURLForKeyword(match.keyword);       \
+  if (turl && turl->prepopulate_id() ==                                       \
+                  TemplateURLPrepopulateData::PREPOPULATED_ENGINE_ID_BRAVE) { \
+    auto raw_gurl = url.spec();                                               \
+    base::ReplaceFirstSubstringAfterOffset(&raw_gurl, 0, "&source=desktop",   \
```

Comment:
this is definitely not the correct way to modify a url. See `GURL::Replacements` to act only on the query string

Reply:
I think there are also some helpers to extract/change specific query string params instead of just a generic find/replace

Reply:
https://source.chromium.org/chromium/chromium/src/+/main:net/base/url_util.h

Reply:
https://source.chromium.org/chromium/chromium/src/+/main:net/base/url_util.h;l=112?q=url_util.h&ss=chromium

Reply:
https://source.chromium.org/chromium/chromium/src/+/main:net/base/url_util.h 

`GetValueForKeyInQuery`, `AppendOrReplaceQueryParameter` with `std::nullopt` and then `AppendQueryParameter` is probably best for changing a query param

Reply:
Ah cool, didn't know about that - looking at the docs I think I can just do `AppendOrReplaceQueryParameter(url, "source", "new_tab")`.